tests/mgr/test_dashboard_url.py

```diff
@@ -0,0 +1,77 @@
+from cli.cephadm.cephadm import CephAdm
+from cli.utilities.utils import get_service_id
+from utility.log import Log
+
+log = Log(__name__)
+
+
+class DashboardUrlError(Exception):
+    pass
+
+
+class CephMgrServiceFailError(Exception):
+    pass
+
+
+class ResourceNotFoundError(Exception):
+    pass
+
+
+def check_dashboard_url(ceph_cluster):
+    """
+    Get dashboard URL from mgr service and check if dashboard URL does not
+    bind to host.containers.internal
+    """
+    url = ceph_cluster.get_mgr_services().get("dashboard")
+    if not url:
+        raise ResourceNotFoundError("Dashboard not configured")
+    if "host.containers.internal" in url:
+        log.error("Dashboard URL {} is not binding to IP of hostname")
+        return False
+    log.info("Dashboard URL is {}".format(url))
+    return True
+
+
+def run(ceph_cluster, **kw):
+    """
+    Check if the Ceph dashboard URL consists of host IP and
+    does not bind to host.containers.internal before and after
+    container restart and mgr failover is performed
+    """
+    # Get mgr node
+    node = ceph_cluster.get_nodes(role="mgr")[0]
+
+    # Check if dashboard URL does not bind to host.containers.internal
+    check_url = check_dashboard_url(ceph_cluster)
+    if check_url is False:
+        raise DashboardUrlError("Dashboard URL binds to host.containers.internal")
+    log.info("Dashboard URL does NOT bind to host.containers.internal")
+
+    # Get container id for mgr daemon
+    out, rc = node.exec_command(sudo=True, cmd="podman ps | grep mgr")
```

Comment:
instead use - https://github.com/red-hat-storage/cephci/blob/master/cli/utilities/utils.py#L60

Reply:
Hey @vamahaja , If I use get_running_containers() method, the output looks as below-
`[
  {
    "AutoRemove": true,
    "Command": [
      "-n",
      "mgr.ceph-saya-patch-yf9uqs-node1-installer.hzpdiz",
      "-f",
      "--setuser",
      "ceph",
      "--setgroup",
      "ceph",
      "--default-log-to-file=false",
      "--default-log-to-journald=true",
      "--default-log-to-stderr=false"
    ],
    "CreatedAt": "27 hours ago",
    "Exited": false,
    "ExitedAt": -62135596800,
    "ExitCode": 0,
    "Id": "12291a94af7435b40ea70bbe93a65b02a6603a441645c9bb0d05de12595eafaa",
    "Image": "registry-proxy.engineering.redhat.com/rh-osbs/rhceph:ceph-6.1-rhel-9-containers-candidate-42152-20230713054431",
    "ImageID": "85cb9476225e77839acf93d726ab3f510398d978d4741d41e846d10ecab19b4e",
    "IsInfra": false,
    "Labels": {
      "CEPH_POINT_RELEASE": "",
      "GIT_BRANCH": "main",
      "GIT_CLEAN": "True",
      "GIT_COMMIT": "0727c855af939c6f3709e73be703026388413744",
      "GIT_REPO": "https://github.com/ceph/ceph-container.git",
      "RELEASE": "main",
      "architecture": "x86_64",
      "build-date": "2023-07-13T05:44:32",
      "ceph": "True",
      "com.redhat.component": "rhceph-container",
      "com.redhat.license_terms": "https://www.redhat.com/agreements",
      "description": "Red Hat Ceph Storage 6",
      "distribution-scope": "public",
      "io.buildah.version": "1.29.0",
      "io.k8s.description": "Red Hat Ceph Storage 6",
      "io.k8s.display-name": "Red Hat Ceph Storage 6 on RHEL 9",
      "io.openshift.expose-services": "",
      "io.openshift.tags": "rhceph ceph",
      "maintainer": "Guillaume Abrioux \u003cgabrioux@redhat.com\u003e",
      "name": "rhceph",
      "release": "192",
      "summary": "Provides the latest Red Hat Ceph Storage 6 on RHEL 9 in a fully featured and supported base image.",
      "url": "https://access.redhat.com/containers/#/registry.access.redhat.com/rhceph/images/6-192",
      "vcs-ref": "dec93361f6f7a22d929d690d9002f0df9a8f6805",
      "vcs-type": "git",
      "vendor": "Red Hat, Inc.",
      "version": "6"
    },
    "Mounts": [
      "/var/lib/ceph/mgr/ceph-ceph-saya-patch-yf9uqs-node1-installer.hzpdiz",
      "/etc/ceph/ceph.conf",
      "/etc/hosts",
      "/var/run/ceph",
      "/var/log/ceph",
      "/var/lib/ceph/crash",
      "/run/systemd/journal"
    ],
    "Names": [
      "ceph-a18090bc-2152-11ee-a7d6-fa163e318b7b-mgr-ceph-saya-patch-yf9uqs-node1-installer-hzpdiz"
    ],
    "Namespaces": {

    },
    "Networks": [],
    "Pid": 1418733,
    "Pod": "",
    "PodName": "",
    "Ports": null,
    "Size": null,
    "StartedAt": 1689584695,
    "State": "running",
    "Status": "Up 27 hours",
    "Created": 1689584695
  }
]`

It does not contain the "container id" which is present in output of `podman ps | grep mgr command`
 -
 
 # podman ps | grep mgr
**12291a94af74**  registry-proxy.engineering.redhat.com/rh-osbs/rhceph:ceph-6.1-rhel-9-containers-candidate-42152-20230713054431                -n mgr.ceph-saya-...  27 hours ago  Up 27 hours              ceph-a18090bc-2152-11ee-a7d6-fa163e318b7b-mgr-ceph-saya-patch-yf9uqs-node1-installer-hzpdiz